rust/lance/src/index/vector/ivf/v2.rs

```diff
@@ -442,52 +543,131 @@ mod tests {
             .await
             .unwrap();
 
-        // TODO: test query after we replace the IVFIndex with the new one
-        // let query = vectors.value(0);
-        // let k = 100;
-        // let result = dataset
-        //     .scan()
-        //     .nearest("vector", query.as_primitive::<Float32Type>(), k)
-        //     .unwrap()
-        //     .nprobs(nlist)
-        //     .with_row_id()
-        //     .try_into_batch()
-        //     .await
-        //     .unwrap();
-
-        // let row_ids = result
-        //     .column_by_name(ROW_ID)
-        //     .unwrap()
-        //     .as_primitive::<UInt64Type>()
-        //     .values()
-        //     .to_vec();
-        // let dists = result
-        //     .column_by_name("_distance")
-        //     .unwrap()
-        //     .as_primitive::<Float32Type>()
-        //     .values()
-        //     .to_vec();
-        // let results = dists
-        //     .into_iter()
-        //     .zip(row_ids.into_iter())
-        //     .collect::<Vec<_>>();
-        // let row_ids = results.iter().map(|(_, id)| *id).collect::<HashSet<_>>();
-
-        // let gt = ground_truth(
-        //     &vectors,
-        //     query.as_primitive::<Float32Type>().values(),
-        //     k,
-        //     DistanceType::L2,
-        // );
-        // let gt_set = gt.iter().map(|r| r.1).collect::<HashSet<_>>();
-
-        // let recall = row_ids.intersection(&gt_set).count() as f32 / k as f32;
-        // assert!(
-        //     recall >= 1.0,
-        //     "recall: {}\n results: {:?}\n\ngt: {:?}",
-        //     recall,
-        //     results,
-        //     gt,
-        // );
+        let query = vectors.value(0);
+        let k = 100;
+        let result = dataset
+            .scan()
+            .nearest("vector", query.as_primitive::<Float32Type>(), k)
+            .unwrap()
+            .nprobs(nlist)
+            .with_row_id()
+            .try_into_batch()
+            .await
+            .unwrap();
+
+        let row_ids = result
+            .column_by_name(ROW_ID)
+            .unwrap()
```

Comment:
FYI, this can be simplified to

```suggestion
        let row_ids = result[ROW_ID]
```